spec/unit/util_spec.rb

```diff
@@ -229,11 +229,29 @@ def get_mode(file)
       (0..256).each {|n| IO.stubs(:new) }
     end
 
-    it "should close all open file descriptors except stdin/stdout/stderr" do
+    it "should close all open file descriptors except stdin/stdout/stderr when /proc/self/fd exists", :if => Dir.exists?('/proc/self/fd') do
```

Comment:
I think we'd rather have `Dir.foreach` stubed for `/proc/self/fd` to return a known list of descriptors, including '.', '..', 0, 1, and 2.  For those special ones, I think we should stub `IO.expects(:new).with(foo).never` so that if it gets call, it will fail the example.